plan/match_property.go

```diff
@@ -25,13 +25,16 @@ func (ts *PhysicalTableScan) matchProperty(prop *requiredProperty, infos ...*phy
 	rowCount := float64(infos[0].count)
 	cost := rowCount * netWorkFactor
 	if len(prop.props) == 0 {
-		return &physicalPlanInfo{p: ts, cost: cost, count: infos[0].count}
+		return enforceProperty(prop, &physicalPlanInfo{p: ts, cost: cost, count: infos[0].count})
```

Comment:
It's no need to check whether prop.limit is nil. If it is, it won't change the plan info.